editor/components/post-pending-status/check.js

```diff
@@ -22,17 +19,11 @@ export function PostPendingStatusCheck( { isPublished, children, user } ) {
 
 export default compose(
 	withSelect( ( select ) => {
-		const { isCurrentPostPublished, getCurrentPostType } = select( 'core/editor' );
+		const { isCurrentPostPublished, getCurrentPostType, getCurrentPost } = select( 'core/editor' );
 		return {
+			hasPublishAction: get( getCurrentPost(), [ '_links', 'wp:action-publish' ], false ),
```

Comment:
I feel there should be a dedicated selector or a mechanism to make this clearer. I see selectors as something that should reflect data needs. I don't have the perfect solution though:

- Something like `canDoActionOnCurrentPost( state, 'publish' )` would work but it's specific to the current post. 
- Just an utility like `canDoAction( record, 'publish' )` could work too but it enforces the fact that the `_links` property is part of the record where ideally it shouldn't be.

Reply:
I think this is covered by, #6655.